apps/frontend/src/app/CreateOrder/CreateOrder.tsx

```diff
@@ -1,51 +1,127 @@
-import React, { FC, useState } from 'react';
+import React, { ChangeEvent, FC, useCallback, useRef, useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { useHistory } from 'react-router-dom';
 
 import type { CreateOrderDto, OrderModel } from '@pasnik/api/data-transfer';
 import { useOrdersFacade } from '@pasnik/orders-data-access';
+import { Alert, Button, InputAdornment, Paper, TextField } from '@mui/material';
+import { formatter, validUrl } from '../../utils';
+
 export const CreateOrder: FC = () => {
   const { register, handleSubmit } = useForm();
   const { createOrder } = useOrdersFacade();
 
+  const [restaurant, setRestaurant] = useState<string>('');
+  const [menu, setMenu] = useState<string>('');
+  const [isValid, setIsValid] = useState<boolean>(false);
+  const [isValidInput, setIsValidInput] = useState<boolean>(false);
+  const [DeliveryPrice, setDeliveryPrice] = useState<string>('');
+
+  const deliveryRef = useRef<HTMLInputElement>(null);
+
   const [error, setError] = useState<string | null>(null);
   const history = useHistory();
 
   const onSubmit = (data: CreateOrderDto) => {
-    createOrder({ ...data, orderAt: new Date().toISOString() })
-      .then((params: OrderModel) => history.push(`/`))
+    const newPrice = data.shippingCents
+      ? +formatter.format(data.shippingCents)
+      : -1;
+    createOrder({
+      ...data,
+      orderAt: new Date().toISOString(),
+      shippingCents: newPrice * 100,
```

Comment:
what if `shippingCents` is not provided? we're passing `-100` here? 🤔

Reply:
I had an idea to add condition in other components    shippingCents>=0 shippingCents : 'shipping price is not added'. This way turned off errors with 'object might be undefined or null'